src/pipeline.cc

```diff
@@ -892,6 +892,14 @@ class PipelineWorker : public Napi::AsyncWorker {
           image.set(s.first.data(), s.second.data());
         }
       }
+      // XMP buffer
+      if (baton->withXmpBuffer != nullptr && baton->withXmpBufferLength > 0) {
+        image = image.copy();
+        image.set(VIPS_META_XMP_NAME, reinterpret_cast<VipsCallbackFn>(vips_area_free_cb),
```

Comment:
nit: Reusing `vips_area_free_cb` doesn't feel right here, the current implementation do the thing that we want https://github.com/libvips/libvips/blob/594187812907bca9499205b8d2e0e35d724f5e3f/libvips/iofuncs/type.c#L137

`g_free` of `mem` without doing anything with the `area`, but nothing guarantees that it will do it in the future. With the size of the implementation I would provide our own free callback (with the added benefit that you will be able to remove the `reinterpret_cast`)

Reply:
the goal is to contribute upstream, so I would assume it's ok to reuse something similar to https://github.com/tpatel/sharp/blob/90bc17db1ab618cdd61e9f028ee99c19710d129f/src/common.cc#L593